lib/events.js

```diff
@@ -750,6 +758,7 @@ EventEmitter.prototype.removeAllListeners =
           else
             delete events[type];
         }
+        this[kShapeMode] = false;
```

Comment:
Why? What if we just have an object template (template in the abstract way not v8 one) and use that when resetting? (Same for the one below)

(This is not really important comment as it's not happening on a regular basis)

Reply:
I don't understand

Reply:
Maybe reset `_events` to the template you provided (e.g. with data event key) instead to the default one  (i.e. empty with proto null)

Reply:
That would require us to keep the template in memory and then create copies. Maybe having some kind of `reset/initEvents` override is better but starts to make things complciated.